internal/literals/random_testing.go

```diff
@@ -0,0 +1,55 @@
+//go:build garble_testing
+
+package literals
+
+import (
+	"go/ast"
+	mathrand "math/rand"
+	"os"
+	"strings"
+
+	"golang.org/x/exp/slices"
+)
+
+var (
+	TestObfuscator         string
+	packageToObfuscatorMap map[string]obfuscator
+)
+
+func init() {
+	obfMapEnv := os.Getenv("GARBLE_TEST_LITERALS_OBFUSCATOR_MAP")
+	if obfMapEnv == "" {
+		panic("literals obfuscator map required for testing build")
+	}
+	packageToObfuscatorMap = make(map[string]obfuscator)
+	// Parse obfuscator map: packageName1=obfName1,packageName2=obfName2
+	pairs := strings.Split(obfMapEnv, ",")
+	for _, pair := range pairs {
+		keyValue := strings.SplitN(pair, "=", 2)
+
+		obfName := keyValue[1]
+		obfIdx := slices.Index(ObfuscatorNames, obfName)
+		if obfIdx < 0 {
+			panic("unknown obfuscator " + obfName)
+		}
+		packageToObfuscatorMap[keyValue[0]] = obfuscators[obfIdx]
+	}
+	TestObfuscator = obfMapEnv
+}
```

Comment:
you could just move this `func init` behind the build tag, and leave everything else in the default build, without any build tags. this means you avoid some code duplication and one extra file.

Reply:
Separate file allows to optimize [this if](https://github.com/burrowers/garble/blob/2546e84e59928fd90d2078c288e7620d86b50aab/hash.go#L149) by declaring const instead of var. If everything is merged into one file, release build will contain code from benchmark

Reply:
I am not worried about one potential branch, to be quite honest. An extra file and dozens of lines of code worry me more :)